### DISCUSSION TITLE
Unable to export data to mail

### DISCUSSION THREAD

**Discussion by shrinidhi111** [outline/outline#2600] [Self Hosting] [answered]
Hi!

I am test driving this tool. Initially, mails did not work at all but that got fixed when I changed the `SMTP_PORT` to 465 as per https://kevalnagda.github.io/sendgrid-with-outline

I created a sample collection with one document and tried exporting it.

But it is not working for me...

Logs is not helping too as there is no errors there..

```
outliner.1.6zorznkatlbu@e2e-75-115    | 2021-09-20T18:21:36.236Z http   <-- POST /api/views.create
outliner.1.6zorznkatlbu@e2e-75-115    | 2021-09-20T18:21:36.287Z http   --> POST /api/views.create 200 51ms 535b
outliner.1.6zorznkatlbu@e2e-75-115    | 2021-09-20T18:21:40.371Z http   <-- GET /api/collections.export?id=60cde51c-5534-442c-a713-ae4ae71ae66c
outliner.1.6zorznkatlbu@e2e-75-115    | 2021-09-20T18:21:40.436Z http   --> POST /api/collections.export?id=60cde51c-5534-442c-a713-ae4ae71ae66c 200 65ms 658b
outliner.1.6zorznkatlbu@e2e-75-115    | 2021-09-20T18:23:59.122Z http   <-- POST /api/documents.viewed
outliner.1.6zorznkatlbu@e2e-75-115    | 2021-09-20T18:23:59.130Z http   <-- POST /api/documents.drafts
outliner.1.6zorznkatlbu@e2e-75-115    | 2021-09-20T18:23:59.134Z http   <-- POST /api/documents.list
```

If you notice `/api/collections.export?id=60cde51c-5534-442c-a713-ae4ae71ae66c 200 65ms 658b` it basically continues without throwing any errors

**To Reproduce**
1. Go to Settings
2. Click on Import/Export
3. Click on "Export Data"
4. I get error in my mail

**Expected behavior**
I should receive the exported information.

**Screenshots**

![image](https://user-images.githubusercontent.com/10372299/134054717-efc246da-4156-42fe-bb2d-454a403acdd1.png)

**Outline (please complete the following information):**
- Install: self hosted
- Version: outlinewiki/outline:latest@sha256:46d4a983b9d98ea5962337b31ec8432b3ba72f891d632cd596619cf0bbf6d7ee

**Desktop (please complete the following information):**

 - Browser 

Google Chrome

**Comment by tommoor** [comment, marked as answer]
If there is an error it will appear in your logs, most likely you don't yet have storage configured? S3 compatible storage must be setup for the export to have somewhere to upload to.

**Reply by shrinidhi111** [reply]
@tommoor 

I had configured S3, roles, permissions as per the guide... But I cross-checked again as per your suggestion and you are right..

The bucket name that I gave in the role had a spelling mistake... Hence uploads didn't happen..

Thanks!

PS :: It'll be useful though, if some error is printed during AWS S3 permission denied..

**Reply by tommoor** [reply]
The error logging is improved in the next version, but you should have seen _something_ in the server logs 🤔

**Reply by shrinidhi111** [reply]
> The error logging is improved in the next version, but you should have seen _something_ in the server logs 🤔

The role I had assigned for the IAM user had a spelling mistake. The bucket name was wrong.

But everything else was correct. Like keys, URL, etc... So maybe it assumed things went well? 🤔 Although access was not actually there?
